community/player.py

```diff
@@ -91,6 +91,58 @@ def update_frustum(self):
 		Frustum.near = normalize(Frustum.near)
 		Frustum.far = normalize(Frustum.far)
 
+	def check_in_frustum_aabb(self, collider):
+		"Frustum check for AABB boxes"
+		planes = (Frustum.left, Frustum.right, Frustum.bottom, Frustum.top, Frustum.near, Frustum.far)
+		result = 2
+
+		if collider.x1 is None or collider.y1 is None or collider.z1 is None \
+				or collider.x2 is None or collider.y2 is None or collider.z2 is None:
+			return 0
+
+		for plane in planes:
+			_in = 0
+			_out = 0
+			normal = plane.xyz
+			w = plane.w
+			if glm.dot(normal, glm.vec3(collider.x1, collider.y1, collider.z1)) + w < 0:
```

Comment:
This can all likely be significantly simplified.
Something like

```py
out = map(lambda v: glm.dot(plane.xyz, v) < -plane.w, [
   glm.vec3(collider.x1, collider.y1, collider.z1),
   glm.vec3(collider.x2, collider.y1, collider.z1),
   ...
   glm.vec3(collider.x1, collider.y2, collider.z2),
])

if all(out):
   return 0

if any(out):
   result = 1 # side note: what do the values for 'result' mean? Could this be made clearer?
```

would be much more elegant. (To be tested, idk if this works :P)

Reply:
alr, lemme test

Reply:
Actually I wanna refactor the frustum check (include colliders for chunks instead of hardcoding impossible to read values and checks)

Reply:
Tbf I copied some of the code (bc it was too verbose to me to retype it) while I was researching into frustum culling and it's not meaningful atm
I think its used to be an indicator if we need to do a more precise search into it, to make an octree/quadtree for example

Reply:
yep its that, if there aren't any then its 2 (compeltely inside)
Otherwise if it intersects with a plane its 1
if its outside its 0

Reply:
It works